service/key/key.go

```diff
@@ -263,19 +265,34 @@ func RemoteWritePasswordKey() string {
 	return "password"
 }
 
-// IsCAPICluster returns true if the cluster is in v1alpha3 and does not have the "azure-operator.giantswarm.io/version" label added by the azure operator.
-// We do not have a provider agnostic label like "giantswarm.io/version" to define this.
+// IsCAPICluster returns true if the cluster has any of the legacy labels such as azure-operator.giantswarm.io/version.
```

Comment:
Are you sure it returns `true` if the cluster has any of the legacy labels? It rather reads like returning `false` below.

Reply:
https://github.com/giantswarm/prometheus-meta-operator/pull/885